### ISSUE TITLE
setting FASTLANE_SKIP_UPDATE_CHECK env variable does not prevent update check

### ISSUE THREAD

**Issue by graemechristie** [fastlane/fastlane#10163]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

[INSERT OUTPUT HERE]

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

[INSERT OUTPUT HERE]

**Comment by graemechristie** [comment]
I am running sigh on a Ci machine (Mac OSX) that does not have net access to the update server

regardless of setting the FASTLANE_SKIP_UPDATE_CHECK environment variable, sigh still attempts to check for updates (and times out).

```
XXXXXXXX:~ xxxxxxxxxxxx$ export FASTLANE_SKIP_UPDATE_CHECK="1"
XXXXXXXX:~ xxxxxxxxxxxx$ FASTLANE_SKIP_UPDATE_CHECK="1" sigh
fastlane could not check for updates error: execution expired

+-------------------------------------+-------+
|           Summary for sigh 2.28.3           |
+-------------------------------------+-------+
| adhoc                               | false |
| development                         | false |
| skip_install                        | false |
| force                               | false |
| ignore_profiles_with_different_name | false |
| skip_fetch_profiles                 | false |
| skip_certificate_verification       | false |
| platform                            | ios   |
+-------------------------------------+-------+
```

**Comment by PetrosIchor** [comment]
Try removing the quotes around `"1"`. Also, unless you are on a very old version of fastlane, your command should be `fastlane sigh`.  If you are on an old version, that update check flag may not have been implimented yet.

**Comment by ohayon** [comment]
Hey @graemechristie! Would you mind sharing the output of `fastlane env` so we can have some more context about the issue? Thank you!! 🚀

**Comment by graemechristie** [comment]
@PetrosIchor I had tried all purmutations of quotes, no quotes, export vs set etc .. that example was just the last iteration. I was using the latest version, which is clearly displayed in the command output I posted.

**Comment by graemechristie** [comment]
```
XXXXXXXXX:~ xxxxxxxxxxxxxxxx$ FASTLANE_SKIP_UPDATE_CHECK=1 fastlane env
fastlane could not check for updates error: execution expired
sh: /Library/Developer/CommandLineTools//usr/bin/xcodebuild: No such file or directory
[12:17:13]: undefined method `split' for nil:NilClass
[12:17:13]: Generating fastlane environment output, this might take a few seconds...
sh: /Library/Developer/CommandLineTools//usr/bin/xcodebuild: No such file or directory
[12:17:13]: undefined method `split' for nil:NilClass
[12:17:13]: Error detecting currently used Xcode installation, please ensure that you have Xcode installed and set it using `sudo xcode-select -s [path]`
[12:17:13]: Could not get Xcode Version
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                |
| --------------------------- | ------------------------------------ |
| OS                          | 10.12.1                              |
| Ruby                        | 2.2.4                                |
| Bundler?                    | false                                |
| Git                         | git version 2.9.3 (Apple Git-75)     |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane  |
| Host                        | Mac OS X 10.12.1 (16B2657)           |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib           |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016           |
| Is contained                | true                                 |
| Is homebrew                 | false                                |
| Is installed via Fabric.app | false                                |
| Xcode Path                  | /Library/Developer/CommandLineTools/ |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |


### fastlane files:

**No Fastfile found**


**No Appfile found**


### fastlane gems

| Gem      | Version | Update-Status  |
| -------- | ------- | -------------- |
| fastlane | 2.50.0  | 💥 Check failed |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| CFPropertyList            | 2.3.5        |
| addressable               | 2.5.1        |
| babosa                    | 1.0.2        |
| bigdecimal                | 1.2.6        |
| bundler                   | 1.14.6       |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| commander-fastlane        | 4.4.5        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.0        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| google-api-client         | 0.12.0       |
| googleauth                | 0.5.1        |
| highline                  | 1.7.8        |
| http-cookie               | 1.0.3        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| io-console                | 0.4.3        |
| json                      | 1.8.1        |
| jwt                       | 1.5.6        |
| libxml-ruby               | 3.0.0        |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| memoist                   | 0.15.0       |
| mime-types                | 3.1          |
| mime-types-data           | 3.2016.0521  |
| mini_magick               | 4.5.1        |
| minitest                  | 5.4.3        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| multipart-post            | 2.0.0        |
| nanaimo                   | 0.2.3        |
| nokogiri                  | 1.5.6        |
| os                        | 0.9.6        |
| plist                     | 3.2.0        |
| power_assert              | 0.2.2        |
| psych                     | 2.0.8        |
| public_suffix             | 2.0.5        |
| rake                      | 10.4.2       |
| rdoc                      | 4.2.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| rouge                     | 1.11.1       |
| rubygems-update           | 2.5.2        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| signet                    | 0.7.3        |
| slack-notifier            | 1.5.1        |
| sqlite3                   | 1.3.13       |
| terminal-notifier         | 1.7.1        |
| terminal-table            | 1.7.3        |
| test-unit                 | 3.0.8        |
| tty-screen                | 0.5.0        |
| uber                      | 0.0.15       |
| unf                       | 0.1.4        |
| unf_ext                   | 0.0.7.4      |
| unicode-display_width     | 1.1.3        |
| word_wrap                 | 1.0.0        |
| xcodeproj                 | 1.4.4        |
| xcpretty                  | 0.2.6        |
| xcpretty-travis-formatter | 0.0.4        |
</details>


*generated on:* **2017-09-01**
</details>
[12:18:13]: 🙄  Wow, that's a lot of markdown text... should fastlane put it into your clipboard, so you can easily paste it on GitHub? (y/n)
y
[12:18:32]: Successfully copied markdown into your clipboard 🎨
[12:18:32]: Open https://github.com/fastlane/fastlane/issues/new to submit a new issue ✅
MAC148100:~ teamcitybuildagent$
```

**Comment by KrauseFx** [comment]
```
XXXXXXXX:~ xxxxxxxxxxxx$ FASTLANE_SKIP_UPDATE_CHECK="1" sigh
```
You seem to be calling a super super old version of fastlane sigh, please prefix the tool by using `fastlane sigh`, that will do the trick. Make sure to run `gem uninstall sigh deliver match` and `gem update fastlane`, or remove all non-fastlane gems from your `Gemfile` if you have one

**Comment by thii** [comment]
I'm getting the same issue. Run a test lane and fastlane only skip checking randomly once every 5 or 10 runs.

**Comment by Lascorbe** [comment]
Also running into the same issue, `ENV["FASTLANE_SKIP_UPDATE_CHECK"]` is `true` in the `before_all` and `after_all` methods, but it doesn't seem to take it into account.

**Comment by joelshea** [comment]
I am also having this issue with 2.63.0.  I have set the environment variable in the Fastfile file, the .env file, and even the Appfile.  I tried "1" and 1 in all these places.  No matter what, the update check is made and the change log is spit out.

**Comment by jrreed** [comment]
Also running into the same issue on 2.69.2. Setting the environment variable in our `.env` file.

**Comment by KrauseFx** [comment]
Ahh, just to clarify: You're all setting the environment variable using a `.env` file that gets loaded by _fastlane_, or do you manually `source` it before running _fastlane_? Could you try manually setting the env variable and see if that works? 

I could imagine we're loading the `.env` file **after** we already sent the update check request, therefore maybe still showing the update message.

Thanks everyone 👍

**Comment by jrreed** [comment]
@KrauseFx, yes that's exactly what I'm doing. I set `FASTLANE_SKIP_UPDATE_CHECK=1` in my `fastlane/.env` file that I was expecting to have been loaded by _fastlane_.

I can confirm that by manually setting `FASTLANE_SKIP_UPDATE_CHECK=1` when running my fastlane command the update check _is_ skipped.

Is this a case where I'm using it wrong? (In which case it might be worth mentioning this in [the docs](https://docs.fastlane.tools/advanced/))

Or is this a bug that will get fixed at some point?

**Comment by chadlwilson** [comment]
Using the environment variables externally to fastlane itself is working okay for me, but I agree it's a bit confusing, and I prefer to minimise this kind of external environment setup in my CI pipelines where possible.

As a side note, it'd be nice if there was an action to use to opt out of this, similar to what we have for opting out of crash reporting and usage - to give a consistent way to turn off all the stuff that might fail in a limited CI environment that might not have unencumbered internet access.

**Comment by KrauseFx** [comment]
Hey everyone, I took a quick look, and I can confirm this is a bug, and it's not something you're doing wrong. 

It's a fastlane bug that can be resolved, the problem is: We start looking for updates [here](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/cli_tools_distributor.rb#L34), and then at a later stage, we load the `.env` files over [here](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/lane_manager.rb#L51). 

This requires some refactoring to make the order correct. It's not something I can do right now, however feel free to jump in a submit a PR for it. I marked the issue as "you can do this" 👍 

For now, there seems to be multiple rather easy workarounds, one of which is to add `FASTLANE_SKIP_UPDATE_CHECK` to your global shell environment. I can see how this might not be 100% ideal, but it should do the job for now. 

I hope that helps, and sorry I won't be able to do the refactor myself right now

**Comment by jrreed** [comment]
This one is hard to test with the _latest_ version of fastlane b/c the output is suppressed when you're _on_ the latest version. However, it is still an issue with `2.79.0` and the [release notes](https://github.com/fastlane/fastlane/releases/tag/2.80.0) in `2.80.0` don't mention a fix

**Comment by joshdholtz** [comment]
Heyyyyyy 👋 Thanks for commenting on this! This will be fixed in `2.86.0` (which should be released early this week) with my PR that loads `dotenv` faster - https://github.com/fastlane/fastlane/pull/12047

Didn't know this issue existed otherwise I would have mentioned it in my PR 😊

**Comment by jrreed** [comment]
This is still broken for me, unfortunately.

**Comment by jrreed** [comment]
Sorry, I guess I should add some more details....

I tried upgrading to `2.86.0`, as well as `2.92.0` (need to be at least on version behind the latest to test), and still received the upgrade notification.

**Comment by joshdholtz** [comment]
@jrreed How are you trying to use `FASTLANE_SKIP_UPDATE_CHECK`?

**Comment by jrreed** [comment]
As mentioned above...

"I set FASTLANE_SKIP_UPDATE_CHECK=1 in my fastlane/.env file that I was
expecting to have been loaded by *fastlane*.

I can confirm that by manually setting FASTLANE_SKIP_UPDATE_CHECK=1 when
running my fastlane command the update check *is* skipped."

**Comment by CihanBoz** [comment]
I have the same problem with 2.89.0! I am using export FASTLANE_SKIP_UPDATE_CHECK=1

**Comment by joshdholtz** [comment]
@jrreed @chnbz Made a fix for this here - #12338
